### PR TITLE
Fill block locations for LocatedFileStatus

### PR DESCRIPTION
**Pull request by nopcoder**
Listing returns LocatedFileStatus which includes file block locations. So far assumed it is not used, but dive into it, spark iterate the block locations for files, and expect an array - null will throw an exception.

### CODE REVIEW

File: clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileStatus.java [hunk 1]
```diff
@@ -12,8 +12,8 @@
  */
 public class LakeFSFileStatus extends FileStatus {
 
-    private String checksum;
-    private String physicalAddress;
+    private final String checksum;
+    private final String physicalAddress;
```
Comment (arielshaqed) [inline clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileStatus.java:16]: Not sure Java value types use final fields. Whatever Hadoop FileStatus does we should copy. (afk...)

File: clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileStatus.java [hunk 2]
```diff
@@ -30,7 +30,7 @@ public String getPhysicalAddress() {
     }
 
     public static class Builder {
-        private Path path;
+        private final Path path;
```
Comment (arielshaqed) [inline clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileStatus.java:33]: Why just this one final field?

**Review by arielshaqed — APPROVED** [review]
Great catch, thanks!

See nonbinding comments about finals.

File: clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileStatus.java (hunk 1 above)
Comment (nopcoder) [reply clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileStatus.java:16]: took the ide recommendations - will revert this file.

File: clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileStatus.java (hunk 2 above)
Comment (nopcoder) [reply clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileStatus.java:33]: guess this one is the one that the builder can't change.

**Pull request merged**